pkg/reconciler/v1alpha1/service/service.go

```diff
@@ -163,34 +162,65 @@ func (c *Reconciler) reconcile(ctx context.Context, service *v1alpha1.Service) e
 
 	service.Status.InitializeConditions()
 
+	config, err := c.config(ctx, service)
+	if err != nil {
+		return err
+	}
+
+	// Update our Status based on the state of our underlying Configuration.
+	service.Status.PropagateConfigurationStatus(&config.Status)
+
+	route, err := c.route(ctx, service)
+	if err != nil {
+		return err
+	}
+
+	// Update our Status based on the state of our underlying Route.
+	ss := &service.Status
+	ss.PropagateRouteStatus(&route.Status)
```

Comment:
seems like there is no need to have `ss` variable here.
We can just use `service.Status.PropagateRouteStatus(&route.Status)`